Sources/AppcuesKit/Presentation/UI/AppcuesView.swift

```diff
@@ -0,0 +1,121 @@
+//
+//  AppcuesView.swift
+//  AppcuesKit
+//
+//  Created by James Ellis on 8/18/22.
+//  Copyright © 2022 Appcues. All rights reserved.
+//
+
+import UIKit
+
+@available(iOS 13.0, *)
+public class AppcuesView: UIView {
+
+    private var isConfigured = false        // appcues instance, embed ID, and view controller set
+    private var isAttachedToView = false    // didMoveToSuperView has been called
+    private var hasTracked = false          // has tracked the embed to check for qualified content
+
+    private var appcues: Appcues?
+    internal private(set) var embedId: String?
+    private weak var viewController: UIViewController?
+
+    // the trait that is controlling this AppcuesView
+    internal var embedTrait: AppcuesEmbedTrait?
+
+    internal weak var experienceController: UIViewController?
+
+    private var heightObserver: NSKeyValueObservation?
+
+    private lazy var preferredHeightConstraint: NSLayoutConstraint = {
+        var constraint = heightAnchor.constraint(equalToConstant: 0)
+        constraint.priority = .defaultLow
+        constraint.isActive = true
+        return constraint
+    }()
+
+    required init?(coder aDecoder: NSCoder) {
+      super.init(coder: aDecoder)
+    }
+
+    internal func configure(appcues: Appcues, embedId: String, viewController: UIViewController) {
+        self.appcues = appcues
+        self.embedId = embedId
+        self.viewController = viewController
+
+        isConfigured = true
+
+        trackEmbed()
+    }
+
+    private func trackEmbed() {
+        // only track if we have not already, the required props have been configured, and we have a valid embed ID
+        guard !hasTracked, isConfigured, isAttachedToView, let embedId = embedId, !embedId.isEmpty, let appcues = appcues else { return }
+        // this screen call notifies the server that the embed is available for rendering content
+        // re-purposing screen for this in the short term, a deeper topic for how to best handle for embeds
+        appcues.screen(title: embedId)
+    }
+
+    override public func didMoveToSuperview() {
+        super.didMoveToSuperview()
+
+        if subviews.isEmpty {
+            // possibly have a delegate for these embeds that would allow the host app to have more control over this?
+            isHidden = true
```

Comment:
Do we need to be forcing the size of `AppcuesView` to zero? `isHidden` is sufficient if embedded in a `UIStackView`, but other places would leave an empty box.

Reply:
yep, that makes sense, updated.

Reply:
setting the height constraint to zero at this point leads to broken behavior with the dynamic sizing for some reason 🤔 working through that one.

Reply:
@mmaatttt yep, thanks.  I did a few experiments on it and pushed up where I've landed so far here.  The goal is to have something that is totally self-contained and dynamic sizing.  It seems like the approach in there now works - basically, have two sets of constraints - the ones used when the content is empty (zero height/width) and the ones used when the content exists and should be dynamically sized ( >= 1 height/width).  Interestingly, the height observation stuff seems totally unnecessary, as the other constraints seem to handle it.